src/hardwareio/IMUCalibDumper/src/IMUCalibDumper.cpp

```diff
@@ -0,0 +1,163 @@
+    gyrFileName = static_cast<std::string>(rf.check("gyrFileName", Value(std::string(std::getenv("HOME"))+"/results/IMUDataDump/gyr.mat")).asString());
+    accFilePath = accFileName;
+    gyrFilePath = gyrFileName;
+    
+    //start the asynchronous and synchronous threads
+    return Thread::start();
+}
+
+double IMUCalibDumper::getPeriod()
+{
+    return 0.1; //period of synchrnous thread
+}
+
+bool IMUCalibDumper::interruptModule()
+{
+    //if the module is asked to stop ask the asynchrnous thread to stop
+    return Thread::stop();
+}
+
+void IMUCalibDumper::onStop()
+{
+    // Create files and save the data
+    if(!fs::exists(accFilePath.parent_path()))
+    {
+        yInfo() << "creating log file folder";
+        fs::create_directories(accFilePath.parent_path());
+    }
+    std::ofstream acclog, gyrlog;
+    acclog.open(accFileName, std::ofstream::out | std::ofstream::trunc);
+    gyrlog.open(gyrFileName, std::ofstream::out | std::ofstream::trunc);
+    acclog << std::scientific;
+    gyrlog << std::scientific; 
+    // dump acc 
+    for(auto vec : accData)
+    {
+       for(auto entry : vec)
+       {
+           acclog << "\t" << entry;
+       }
+       acclog << std::endl;
+    }
+    acclog.close();
+
+    // dump gyro
+    for(auto vec : gyrData)
+    {
+       for(auto entry : vec)
+       {
+           gyrlog << "\t" << entry;
+       }
+       gyrlog << std::endl;
+    }
+    gyrlog.close();
+    
+    // clean up
+    imu_port.close();
+}
+
+//synchronous thread
+bool IMUCalibDumper::updateModule()
+{
+    return Thread::isRunning();
+}
+
+//asynchronous thread run forever
+void IMUCalibDumper::run()
+{
+    Stamp yarpstamp_imu;
+    DEF_SENSORS(); // SENSORS keeps track of which sensors are set
+    double t0;
+    bool firstTime = true;
+
+    while(true) {
+        const vector<IMUevent> * q_imu = imu_port.read(yarpstamp_imu);
+        if(!q_imu || Thread::isStopping()) return;
+
+
+        for( ev::IMUevent e_imu : *q_imu)
+        {
+            SET_SENSOR(e_imu.sensor); // set sensor e_imu.sensor as received
+            imu_state[e_imu.sensor] =  e_imu.value;
+                
+            // continue receiving data until the 10th reading (index 9)
+            if(e_imu.sensor != 9) continue;
+
+            // If the 10 sensors were not received, restarts gathering the data
+            if(SENSORS != ALL_SENSORS)
+            {
+                RESET_SENSORS();
+                continue;
+            }
+
+            //resets for the next iteration
+            RESET_SENSORS();
+            
+            vtsHelper(e_imu.stamp);
+            double currTime = static_cast<double>(vtsHelper.currentTime()*vtsHelper.tsscaler);
+            if(firstTime)
+            {
+                t0 = currTime;
+                firstTime = false;
+            }
+
+            std::vector<double> dacc, dgyr;
+
+            dacc.push_back(currTime-t0);
```

Comment:
it would be useful to save also the yarp timestamp with each line of data, to future proof, as it is possible to synchronise this data with other data if needed in the future.

however, if the data file needs to be very specific here, to load into the calibration software, ignore this